server/public/plugin/client_rpc.go

```diff
@@ -358,9 +359,39 @@ func init() {
 	hookNameToId["ServeHTTP"] = ServeHTTPID
 }
 
+// Using a subset of http.Request prevents a known incompatibility when decoding Go v1.23+ gob-encoded x509.Certificate
+// structs from Go v1.22 compiled plugins. These come from http.Request.TLS field (*tls.ConnectionState).
```

Comment:
Can we plan to capture the omission of the `TLS` field altogether? I don't know of a use case for a plugin looking at this, but if we find one I imagine we can expose a subset of that struct if needed.

Reply:
@lieut-data To clarify, are you asking to be more explicit and potentially document the omission further?

The TLS field was [effectively omitted](https://github.com/mattermost/mattermost/blob/ad73ef7340bd27489efd1602c56aa2287d9856aa/server/public/plugin/client_rpc.go#L405-L415) before this PR, along with any other field we are not explicitly passing to the plugin side. The difference is purely in the type we marshal on the server side. Practically this makes no difference as the receiver will still be handling (and expect) a `http.Request` struct with those fields at their default (empty) state.

Reply:
Oh! I missed that -- then no concerns at all here. 🎉